doc/source/feature_overview.md

```diff
@@ -0,0 +1,263 @@
+---
+file_format: mystnb
+jupytext:
+  formats: md:myst
+  text_representation:
+    extension: .md
+    format_name: myst
+kernelspec:
+  display_name: geoutils-env
+  language: python
+  name: geoutils
+---
+(feature-overview)=
+
+# Feature overview
+
+The following presents a descriptive example show-casing all core features of GeoUtils.
+
+For more details, refer to the {ref}`core-index`, {ref}`rasters-index` or {ref}`vectors-index` pages.
+
+```{tip}
+All pages of this documentation containing code cells can be **run interactively online without the need of setting up your own environment**. Simply click the top launch button!
+(MyBinder can be a bit capricious: you might have to be patient, or restart it after the build is done the first time 😅)
+
+Alternatively, start your own notebook to test GeoUtils at [![Binder](https://mybinder.org/badge_logo.svg)](https://mybinder.org/v2/gh/GlacioHack/geoutils/main).
+```
+
+## The core {class}`~geoutils.Raster` and {class}`~geoutils.Vector` classes
+
+In GeoUtils, geospatial handling is object-based and revolves around {class}`~geoutils.Raster` and {class}`~geoutils.Vector`.
+These link to either **on-disk** or **in-memory** datasets, opened by instantiating the class with a file path.
```

Comment:
This business of classes and objects, composition classes is perhaps already a bit detailed for this point in the Features overview. The word 'instantiatiing' is also a bit l33t ;) I'd consider simplifying this section.

Reply:
Agreed! (my head was too deep in the code when I wrote this haha)
I've changed all occurences of "class" into "object", "instantiate" into "call", and removed notions of "composition" or "subclassing".